addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -24,13 +32,21 @@ const GalleryWidget = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onClickImg: function (ev) {
-        if (this.$modal || ev.currentTarget.matches("a > img")) {
+        const clickedEl = ev.currentTarget;
+        if (this.$modal || clickedEl.matches("a > img")) {
             return;
         }
         var self = this;
-        var $cur = $(ev.currentTarget);
 
-        var $images = $cur.closest('.s_image_gallery').find('img');
+        const imageGalleryEl = clickedEl.closest(".s_image_gallery");
+        let imageEls = imageGalleryEl.querySelectorAll("img");
+        const currentImageEl = (clickedEl.matches("img") === true) ? clickedEl : clickedEl.closest("img");
+        const currentImageIndex = [...imageEls].indexOf(currentImageEl);
+        imageEls = [...imageEls].map(el => el.cloneNode(true));
+        // We need to reset the current image to its original source because it
+        // might have been changed by a mouse event (e.g. "hover effect" animation).
+        imageEls[currentImageIndex].src = this.originalSources[currentImageIndex];
```

Comment:
I wonder if it wouldn't be safer to do this for all images in the gallery. :thinking:

Reply:
Yes maybe, I was thinking about doing it too. I will do it in the case an hover animation takes too long time to complete...